cni-plugins/sriov-passthrough-cni/plugin/sriov-passthrough-cni

```diff
@@ -23,64 +22,30 @@ function main() {
     esac
 }
 
-function flip_vfs_to_default_drivers() {
-    for file in $(find /sys/devices/ -name *sriov_totalvfs*); do
-        pfroot=$(dirname $file)
-
-        # flip all available VFs to default driver
-        echo 0 > $pfroot/sriov_numvfs
-        cat $file > $pfroot/sriov_numvfs
-    done
-}
-
-function from_cni_args() {
-    local var_name="${1:?}"
-    sed -nE "s/.+?${var_name}=([^;]*?)(;.+?|$)/\1/p" <<< "$CNI_ARGS"
-}
-
 function setup_netns() {
     mkdir -p /var/run/netns
     ln -sfT "$CNI_NETNS" "$CONTAINER_NETNS_LINK"
 }
 
-function setns_sriov_ifs() {
-  local sriov_vfs=( /sys/class/net/*/device/virtfn* )
-  local ifs_arr ifs_name
-  for vf in "${sriov_vfs[@]}"; do
-    ifs_arr=( "$vf"/net/* )
-    for ifs in "${ifs_arr[@]}"; do
-        ifs_name="${ifs%%\/net\/*}"
-        ifs_name="${ifs##*\/}"
-        log "Adding vf: $ifs_name to netns $CNI_CONTAINERID"
-        ip link set "$ifs_name" netns "$CNI_CONTAINERID"
-    done
-  done
+function setns_sriov_pfs() {
```

Comment:
We have to block this function with a mutex or the like on the node.

Reply:
we don't need, its re-entrant because a PF can exists only on one NS, so it checks if it managed to move the PF,
if it didn't it continue to the next one.

Reply:
Two jobs running at the same time can end up with the same sriov_pfs list so one of them will fail at ip link set and the job will fail, if we use a mutex at node somehow one of them will just wait a little.

Reply:
The list will be the same, but It won't fail, because there are more PFs which will be enough for each schedule job,
it will just detect the collision and allocate the next one.
We won't spin more than available amount of PFs
see please https://github.com/kubevirt/project-infra/pull/784
This is the design
Also mentioned it in the PR description.

Reply:
Ahh right it will go to the next PF and that's all.